### PR TITLE
webgpu: Fix the shared memory exceeds limit in some situation

### PR DESCRIPTION
**Pull request by qjia7**
BUG

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4807)

### CODE REVIEW

**Review by haoyunfeix — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4807)* status: 0 of 1 approvals obtained

**Comment by qjia7** [comment]
@kainino0x @lina128 Please take a look, thanks.

File: tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts [hunk 1]
```diff
@@ -132,6 +132,7 @@ export class MatMulPackedVec4Program implements WebGPUProgram {
     this.dispatchLayout = {x: [2], y: [1], z: [0]};
     const vecSize = 4;
     if (outputShape[1] === 1) {
+      this.workGroupSize = [16, 1, 1];  // in case shared memory exceeds limit.
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts:135]: Previously, when `outputShape[1] = 1`, the workGroupSize will be [64, 1, 1]. For the vec4 matmul, `mm_Bsub` will become `vec4 mm_Bsub[64*4][64]` which is  64K exceeding the limit 32K on windows.

**Comment by qjia7** [comment]
@kainino0x @lina128 Please take a look, thanks.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4807#-:-MVblyc51Nh-ihRQyIRs:bb6ywmx)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @kainino0x)

**Review by kainino0x — APPROVED** [review]

**Pull request merged**